tools/file_packager.py

```diff
@@ -289,6 +289,12 @@ def main():
   }
   Module.expectedDataFileDownloads++;
   (function() {
+   // When running as a pthread, FS operations are proxied to the main thread, so we don't need to
+   // fetch the .data bundle on the worker
+   if (Module['ENVIRONMENT_IS_PTHREAD']) {
+     if (!Module['preRun']) Module['preRun'] = [];
```

Comment:
I'm not sure what you are trying do here.. it looks like you are setting this to the empty list if and only if its already empty.

Also, `preRun` callbacks are only ever run once on the main thread.     See here where we call preRun there is an early return if we are running on a thread so we never get here:  https://github.com/emscripten-core/emscripten/blob/ae5c256cacb32b1a0f303566ee7ac18448a9d7fc/src/postamble.js#L280

Reply:
Hmm, I added this logic because the generated glue file was calling Array.slice on `Module['preRun']` (and so used the code from https://github.com/emscripten-core/emscripten/blob/main/tools/file_packager.py#L876 ), 

But I no longer see it with version 3.0.0. Maybe something changed around his, or I had some local changes to my toolchain. Will revert.